CMake Error when installation in Colab

## 🐛 Bug
```
CMake Error: The source directory "/content/gdrive/My Drive/habitat-testing/habitat-sim/Drive/habitat-testing/habitat-sim/src" does not exist.
Specify --help for usage, or press the help button on the CMake GUI.
```
after running
```
!python setup.py install
```

## Commands
Command
```
# Mount your google drive
from google.colab import drive
drive.mount('/content/gdrive')
```
Output
```
Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount("/content/gdrive", force_remount=True).
```
Command
```
cd 'gdrive/My Drive/habitat-testing'
```
Command
```
!git clone --branch stable https://github.com/facebookresearch/habitat-sim.git
```
Output
```
Cloning into 'habitat-sim'...
remote: Enumerating objects: 736, done.
remote: Counting objects: 100% (736/736), done.
remote: Compressing objects: 100% (313/313), done.
remote: Total 7126 (delta 531), reused 546 (delta 413), pack-reused 6390
Receiving objects: 100% (7126/7126), 31.32 MiB | 14.52 MiB/s, done.
Resolving deltas: 100% (5120/5120), done.
Note: checking out '00cddeee6d66108e277dc5b36006bf3eab03a291'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

Checking out files: 100% (380/380), done.
```
Command
```
cd 'habitat-sim'
```
Output
```
/content/gdrive/My Drive/habitat-testing/habitat-sim
```
Command
```
pip install -r requirements.txt
```
Output
```
Requirement already satisfied: attrs>=19.1.0 in /usr/local/lib/python3.6/dist-packages (from -r requirements.txt (line 1)) (19.3.0)
Requirement already satisfied: numba in /usr/local/lib/python3.6/dist-packages (from -r requirements.txt (line 2)) (0.47.0)
Requirement already satisfied: numpy in /usr/local/lib/python3.6/dist-packages (from -r requirements.txt (line 3)) (1.17.5)
Collecting numpy-quaternion
  Downloading https://files.pythonhosted.org/packages/a6/1e/62ee5115236aafa3d763226da9ad01abc91b8be9cfb843aa0f709ec38f5d/numpy-quaternion-2019.12.11.22.25.52.tar.gz (50kB)
     |████████████████████████████████| 51kB 3.4MB/s 
Requirement already satisfied: pillow in /usr/local/lib/python3.6/dist-packages (from -r requirements.txt (line 5)) (6.2.2)
Requirement already satisfied: scipy>=1.3.0 in /usr/local/lib/python3.6/dist-packages (from -r requirements.txt (line 6)) (1.4.1)
Requirement already satisfied: tqdm in /usr/local/lib/python3.6/dist-packages (from -r requirements.txt (line 7)) (4.28.1)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/dist-packages (from numba->-r requirements.txt (line 2)) (45.1.0)
Requirement already satisfied: llvmlite>=0.31.0dev0 in /usr/local/lib/python3.6/dist-packages (from numba->-r requirements.txt (line 2)) (0.31.0)
Building wheels for collected packages: numpy-quaternion
  Building wheel for numpy-quaternion (setup.py) ... done
  Created wheel for numpy-quaternion: filename=numpy_quaternion-2019.12.11.22.25.52-cp36-cp36m-linux_x86_64.whl size=153317 sha256=05ad13fc1dce0bc143ffc360a3d1ac6ccc8c39f4b19a25c4a08234e03bd89118
  Stored in directory: /root/.cache/pip/wheels/3f/16/3d/a49dc001d0ba4873afeb18803a2e4bc7a3a99a4d9ed65a1b55
Successfully built numpy-quaternion
Installing collected packages: numpy-quaternion
Successfully installed numpy-quaternion-2019.12.11.22.25.52
```
Command
```
!python setup.py install --headless
```
Output
```
running install
running bdist_egg
running egg_info
writing habitat_sim.egg-info/PKG-INFO
writing dependency_links to habitat_sim.egg-info/dependency_links.txt
writing requirements to habitat_sim.egg-info/requires.txt
writing top-level names to habitat_sim.egg-info/top_level.txt
reading manifest template 'MANIFEST.in'
writing manifest file 'habitat_sim.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_py
running build_ext
CMake Error: The source directory "/content/gdrive/My Drive/habitat-testing/habitat-sim/Drive/habitat-testing/habitat-sim/src" does not exist.
Specify --help for usage, or press the help button on the CMake GUI.
Traceback (most recent call last):
  File "setup.py", line 380, in <module>
    include_package_data=True,
  File "/usr/local/lib/python3.6/dist-packages/setuptools/__init__.py", line 145, in setup
    return distutils.core.setup(**attrs)
  File "/usr/lib/python3.6/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.6/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/usr/local/lib/python3.6/dist-packages/setuptools/command/install.py", line 67, in run
    self.do_egg_install()
  File "/usr/local/lib/python3.6/dist-packages/setuptools/command/install.py", line 109, in do_egg_install
    self.run_command('bdist_egg')
  File "/usr/lib/python3.6/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/usr/local/lib/python3.6/dist-packages/setuptools/command/bdist_egg.py", line 172, in run
    cmd = self.call_command('install_lib', warn_dir=0)
  File "/usr/local/lib/python3.6/dist-packages/setuptools/command/bdist_egg.py", line 158, in call_command
    self.run_command(cmdname)
  File "/usr/lib/python3.6/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/usr/local/lib/python3.6/dist-packages/setuptools/command/install_lib.py", line 11, in run
    self.build()
  File "/usr/lib/python3.6/distutils/command/install_lib.py", line 109, in build
    self.run_command('build_ext')
  File "/usr/lib/python3.6/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "setup.py", line 202, in run
    self.build_extension(ext)
  File "setup.py", line 269, in build_extension
    env=env,
  File "/usr/lib/python3.6/subprocess.py", line 311, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '-H/content/gdrive/My', 'Drive/habitat-testing/habitat-sim/src', '-Bbuild', '-DBUILD_PYTHON_BINDINGS=ON', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/content/gdrive/My Drive/habitat-testing/habitat-sim/build/lib.linux-x86_64-3.6/habitat_sim/_ext', '-DPYTHON_EXECUTABLE=/usr/bin/python3', '-DCMAKE_EXPORT_COMPILE_COMMANDS=ON', '-DCMAKE_BUILD_TYPE=RelWithDebInfo', '-DBUILD_GUI_VIEWERS=OFF', '-DTARGET_HEADLESS=ON', '-DBUILD_TEST=OFF', '-DBUILD_WITH_BULLET=OFF', '-DBUILD_DATATOOL=OFF', '-DBUILD_WITH_CUDA=OFF']' returned non-zero exit status 1.
```
Command
```
!python setup.py install
```
Output
```
running install
running bdist_egg
running egg_info
writing habitat_sim.egg-info/PKG-INFO
writing dependency_links to habitat_sim.egg-info/dependency_links.txt
writing requirements to habitat_sim.egg-info/requires.txt
writing top-level names to habitat_sim.egg-info/top_level.txt
reading manifest template 'MANIFEST.in'
writing manifest file 'habitat_sim.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_py
running build_ext
CMake Error: The source directory "/content/gdrive/My Drive/habitat-testing/habitat-sim/Drive/habitat-testing/habitat-sim/src" does not exist.
Specify --help for usage, or press the help button on the CMake GUI.
Traceback (most recent call last):
  File "setup.py", line 380, in <module>
    include_package_data=True,
  File "/usr/local/lib/python3.6/dist-packages/setuptools/__init__.py", line 145, in setup
    return distutils.core.setup(**attrs)
  File "/usr/lib/python3.6/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.6/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/usr/local/lib/python3.6/dist-packages/setuptools/command/install.py", line 67, in run
    self.do_egg_install()
  File "/usr/local/lib/python3.6/dist-packages/setuptools/command/install.py", line 109, in do_egg_install
    self.run_command('bdist_egg')
  File "/usr/lib/python3.6/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/usr/local/lib/python3.6/dist-packages/setuptools/command/bdist_egg.py", line 172, in run
    cmd = self.call_command('install_lib', warn_dir=0)
  File "/usr/local/lib/python3.6/dist-packages/setuptools/command/bdist_egg.py", line 158, in call_command
    self.run_command(cmdname)
  File "/usr/lib/python3.6/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/usr/local/lib/python3.6/dist-packages/setuptools/command/install_lib.py", line 11, in run
    self.build()
  File "/usr/lib/python3.6/distutils/command/install_lib.py", line 109, in build
    self.run_command('build_ext')
  File "/usr/lib/python3.6/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "setup.py", line 202, in run
    self.build_extension(ext)
  File "setup.py", line 269, in build_extension
    env=env,
  File "/usr/lib/python3.6/subprocess.py", line 311, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '-H/content/gdrive/My', 'Drive/habitat-testing/habitat-sim/src', '-Bbuild', '-DBUILD_PYTHON_BINDINGS=ON', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/content/gdrive/My Drive/habitat-testing/habitat-sim/build/lib.linux-x86_64-3.6/habitat_sim/_ext', '-DPYTHON_EXECUTABLE=/usr/bin/python3', '-DCMAKE_EXPORT_COMPILE_COMMANDS=ON', '-DCMAKE_BUILD_TYPE=RelWithDebInfo', '-DBUILD_GUI_VIEWERS=ON', '-DTARGET_HEADLESS=OFF', '-DBUILD_TEST=OFF', '-DBUILD_WITH_BULLET=OFF', '-DBUILD_DATATOOL=OFF', '-DBUILD_WITH_CUDA=OFF']' returned non-zero exit status 1.
```

## System Info
Google Colab.

--- a/setup.py
+++ b/setup.py
@@ -264,7 +264,9 @@ class CMakeBuild(build_ext):
 
         if self.run_cmake(cmake_args):
             subprocess.check_call(
-                shlex.split('cmake -H"{}" -B"{}"'.format(ext.sourcedir, self.build_temp))
+                shlex.split(
+                    'cmake -H"{}" -B"{}"'.format(ext.sourcedir, self.build_temp)
+                )
                 + cmake_args,
                 env=env,
             )

--- a/setup.py
+++ b/setup.py
@@ -264,7 +264,7 @@ class CMakeBuild(build_ext):
 
         if self.run_cmake(cmake_args):
             subprocess.check_call(
-                shlex.split("cmake -H\"{}\" -B\"{}\"".format(ext.sourcedir, self.build_temp))
+                shlex.split('cmake -H"{}" -B"{}"'.format(ext.sourcedir, self.build_temp))
                 + cmake_args,
                 env=env,
             )
@@ -273,7 +273,7 @@ class CMakeBuild(build_ext):
             self.create_compile_commands()
 
         subprocess.check_call(
-            shlex.split("cmake --build \"{}\"".format(self.build_temp)) + build_args
+            shlex.split('cmake --build "{}"'.format(self.build_temp)) + build_args
         )
         print()  # Add an empty line for cleaner output
 

--- a/setup.py
+++ b/setup.py
@@ -264,7 +264,7 @@ class CMakeBuild(build_ext):
 
         if self.run_cmake(cmake_args):
             subprocess.check_call(
-                shlex.split("cmake -H{} -B{}".format(ext.sourcedir, self.build_temp))
+                shlex.split("cmake -H\"{}\" -B\"{}\"".format(ext.sourcedir, self.build_temp))
                 + cmake_args,
                 env=env,
             )
@@ -273,7 +273,7 @@ class CMakeBuild(build_ext):
             self.create_compile_commands()
 
         subprocess.check_call(
-            shlex.split("cmake --build {}".format(self.build_temp)) + build_args
+            shlex.split("cmake --build \"{}\"".format(self.build_temp)) + build_args
         )
         print()  # Add an empty line for cleaner output
 

